lib/CL/devices/devices.c

```diff
@@ -135,13 +142,15 @@ const char *
 pocl_get_device_name (unsigned index)
 {
 
-  if (index < pocl_num_devices)
+  if (index < POCL_ATOMIC_LOAD (pocl_num_devices))
```

Comment:
Does atomic load suffice here? Don't we need to lock the entire device list as it might change before you enter the traversal code?

Reply:
Currently, device list is append only, so locking is not required.